### ISSUE TITLE
Regarding Chinese languages on Weblate

### ISSUE THREAD

**Issue by akien-mga** [godotengine/godot#21314]
I'd like input from Chinese speaking contributors regarding the various Chinese locales we have up for independent translation on https://hosted.weblate.org/projects/godot-engine/godot

As of today we have 6 different Chinese translations being worked on in parallel, and I think there might be some confusion regarding what is what, so I'd like input from people actually familiar with those languages.

Right now we have in Godot:

- `zh_CN` (`zh_Hans` in Windows API and on Weblate) - Chinese (Simplified)
- `zh_TW` (`zh_Hant` in Windows API/Weblate) - Chinese (Traditional)
- `zh_HK` (`zh_Hant_HK` in Windows API/Weblate) - Chinese (Hong Kong)

Those three I'm familiar with, and they make sense to be separate as they are wholly different.

But recently translators added on Weblate:

- `zh` - Chinese
  Does this make sense to have? There's no absolute "Chinese" that I know of.
- `zh_CHS` - Chinese (CHS)
  This seems to be the old Windows naming for zh_CN/zh_Hans, so seems redundant with Chinese (Simplified)
- `cmn` - Chinese (Mandarin)
  Here I'm lost, help :) In Godot's known locales we do have "cnm_TW" for "Mandarin Chinese (Taiwan)". Can you confirm that it's a different dialect from zh_TW and should be kept separate?

For the reference, here are the "Chinese" locales currently recognized by Godot:
```
core/translation.cpp
109:    "cmn_TW", //  Mandarin Chinese (Taiwan)
201:    "hak_TW", //  Hakka Chinese (Taiwan)
254:    "lzh_TW", //  Literary Chinese (Taiwan)
275:    "nan_TW", //  Min Nan Chinese (Taiwan)
396:    "yue_HK", //  Yue Chinese (Hong Kong)
397:    "zh", //  Chinese
398:    "zh_CN", //  Chinese (China)
399:    "zh_HK", //  Chinese (Hong Kong)
400:    "zh_SG", //  Chinese (Singapore)
401:    "zh_TW", //  Chinese (Taiwan)
```

Thanks for any input!

**Comment by akien-mga** [comment]
CC @Geequlim @sersoong @ZX-WT

**Comment by akien-mga** [comment]
> - `zh_CHS` - Chinese (CHS)
> This seems to be the old Windows naming for zh_CN/zh_Hans, so seems redundant with Chinese (Simplified)

Should I file this as a bug with Weblate @nijel? I guess `zh_CHS` should be marked as a variant of `zh_Hans`?

**Comment by nijel** [comment]
There are some duplicates in Weblate language definitions indeed, see https://github.com/WeblateOrg/weblate/issues/2072. I'll look into that now.

**Comment by nijel** [comment]
I've just done massive cleanup. The biggest part of the problem comes from people using various codes for same language and Weblate will never be able to detect all variants. I've updated the mappings to match current knowledge and removed duplicate ones.

I now recommend you to remove the additional translation files from the vcs, sorry for the mess.

**Comment by akien-mga** [comment]
Thanks!

**Comment by sersoong** [comment]
I think all languages of Taiwan area(TW) can be merged into zh_TW.

Although there are many dialects in Taiwan, only one official language(zh_TW) can be used,the rest is meaningless.

Chinese (Mandarin) is the standard language in mainland China.I mean **pronunciation**.
Mandarin Chinese (Taiwan) is the common language in taiwan China.
It is a **little different** from the mainland of China.

The language of Hongkong and Macao is the same as that of Guangdong of China.Because of historical reasons, they have some subtle differences.They use Cantonese(Yue Chinese).

Yue Chinese, Min Nan Chinese and Hakka Chinese is a minority language. 

**Subtle cultural and linguistic differences** lead to different words.

From the perspective of writing, there are differences between simplified Chinese and traditional Chinese.

Of course, there are other Chinese characters in Japanese.But I am not discuss this.

**Differences in writing** 

simplified Chinese = zh_CHS = zh_Hans = zh_SG
traditional Chinese = zh_CHT = zh_TW = zh_HK
But zh_HK is different from  zh_TW.

**Subtle cultural and linguistic differences** lead to different words.
So,I think you can keep zh_CN zh_TW zh_HK on weblate.
"zh" is no sense.Usually it is used to represent the Chinese mainland, that is, zh_CN.

**Comment by akien-mga** [comment]
Thanks for the detailed input @sersoong, that confirms that @nijel did the right changes.
